src/main/java/bio/terra/service/filedata/flight/ingest/ValidateIngestFileDirectoryStep.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.model.FileLoadModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.filedata.exception.FileAlreadyExistsException;
+import bio.terra.service.filedata.exception.FileSystemAbortTransactionException;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.service.filedata.google.firestore.FireStoreDirectoryEntry;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.stairway.*;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ValidateIngestFileDirectoryStep implements Step {
+    private static final Logger logger = LoggerFactory.getLogger(ValidateIngestFileDirectoryStep.class);
+
+    private final FireStoreDao fileDao;
+    private final Dataset dataset;
+
+    public ValidateIngestFileDirectoryStep(FireStoreDao fileDao,
+                                  Dataset dataset) {
+        this.fileDao = fileDao;
+        this.dataset = dataset;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) throws InterruptedException {
+        FlightMap inputParameters = context.getInputParameters();
+        FileLoadModel loadModel = inputParameters.get(JobMapKeys.REQUEST.getKeyName(), FileLoadModel.class);
+
+        String targetPath = loadModel.getTargetPath();
+
+        try {
+            //  If the directory entry exists and the loadTags do not match,
+            //  then we throw FileAlreadyExistsException.
+            FireStoreDirectoryEntry existingEntry = fileDao.lookupDirectoryEntryByPath(dataset, targetPath);
+            if (existingEntry != null && !StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
```

Comment:
Can a load tag ever be null? If not, the use of `StringUtils` can be avoided here
```suggestion
            if (existingEntry != null && !existingEntry.getLoadTag().equals(loadModel.getLoadTag())) {
```

Reply:
I like the defensive code there...seems safer to use StringUtils there in the off chance that it does end up being null

Reply:
But it's not possible for a load tag to be null, is it?

Reply:
Oh I see what you mean, the `FileLoadModel` sets the default value to an empty string so this shouldn't be null